charmhelpers/fetch/ubuntu.py

```diff
@@ -621,3 +621,7 @@ def get_upstream_version(package):
         return None
 
     return apt_pkg.upstream_version(pkg.current_ver.ver_str)
+
+
+def get_distrib_codename():
```

Comment:
This is logically in the wrong file.  I think it should be imported from charmhelpers.core.host like `lsb_release()` is, which, yes, does mean implementing it in the host_factory/ubuntu.py and then bubbling it through the imports.

Also, a docstring is needed.  It needs something like:

"""Return the codename of the distribution

:returns: The codename
:rtype: str
"""